### DISCUSSION TITLE
Sphinx confused by aliases/How to not get aliases in documentation.

### DISCUSSION THREAD

**Discussion by ajoino** [pydantic/pydantic#2514] [Question] [answered]
Hi.
First of all, I would just like to say that I really love this project, I was halfway into implementing many features of Pydantic when I discovered it, and it saved me much time and effort.
However, there are some quirks that I haven't figured out how to fix, and one of them is related to Sphinx and docstrings, namely that the autogenerated documentation will use the aliases of the fields instead of the field names. Is there a setting that makes Pydantic use the regular field names in the docstring instead of the alias?

The use case is that I'm writing a library that acts as an interface with a framework, but it is not important for the users to know that the framework's API uses camelCase. I know I could just add `(str)` in the docstring manually for those fields but it would be nice to have an automated solution for this.

Kind regards
Jacob

**Comment by samuelcolvin** [comment, marked as answer]
Thanks for your kind words.

Good question, I'm afraid the short answer is "no".

The longer answer is ...

The logic for generating the signature is

https://github.com/samuelcolvin/pydantic/blob/619ff261c9bdc2a612ef2fcfa17a005dfefbcf21/pydantic/utils.py#L218-L227

it always uses the field alias unless that alias is not a valid identifier.

Your options to change this are:
* implement your own `generate_model_signature` and add a custom docstring to your models, not sure how easy this would be in a general way, but I imagine not impossible
* use a hack like using invalid identifiers for aliases
* if other people are keen on this, I guess we could add a config flag to change this logic, but I'd rather not if possible

**Reply by ajoino** [reply]
Thanks for the quick and detailed answer.

I accept the small amount of confusion, a good docstring should be enough to clear any confusion that may occur.
